pkg/app/web/src/components/application-detail-page/application-state-view/cloudrun-state-view/index.tsx

```diff
@@ -0,0 +1,196 @@
+import { Box, makeStyles } from "@material-ui/core";
+import clsx from "clsx";
+import dagre from "dagre";
+import { FC, useState } from "react";
+import { CloudRunResourceState } from "~/modules/applications-live-state";
+import { theme } from "~/theme";
+import { uniqueArray } from "~/utils/unique-array";
+import { CloudRunResource } from "./cloudrun-resource";
+import { CloudRunResourceDetail } from "./cloudrun-resource-detail";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    flex: 1,
+    justifyContent: "center",
+    overflow: "hidden",
+  },
+  stateViewWrapper: {
+    flex: 1,
+    display: "flex",
+    justifyContent: "center",
+    overflow: "hidden",
+  },
+  stateView: {
+    position: "relative",
+    overflow: "auto",
+  },
+  closeDetailButton: {
+    position: "absolute",
+    right: theme.spacing(1),
+    top: theme.spacing(1),
+    color: theme.palette.grey[500],
+  },
+}));
+
+const getServiceID = (resources: CloudRunResourceState.AsObject[]): string => {
+  for (let resource of resources) {
+    if (resource.parentIdsList.length === 0) return resource.id;
+  }
+  return "";
+}
```

Comment:
Please give me advise to refactor this function if you feel uncomfortable.

Reply:
I think we can use filter instead of for loop like this, and since this function is only be used in a single place, should not extract as a function, directly use filter to find out serviceID is good enough 👀

Reply:
https://github.com/pipe-cd/pipecd/pull/3301/files#r811540126